### PR TITLE
{python3_05} Python3: fix masks, logging and range

### PR DESCRIPTION
**Pull request by cjmayo**

### CODE REVIEW

**Comment by cjmayo** [comment]
Rebased onto python3_04 to attempt to pass tests (missing builtins module). I will wait to fix the other PR test failures until this one is working.

File: linkcheck/configuration/__init__.py [hunk 1]
```diff
@@ -403,7 +403,7 @@ def make_userdir(child):
             # Windows forbids filenames with leading dot unless
             # a trailing dot is added.
             userdir += "."
-        os.makedirs(userdir, 0700)
+        os.makedirs(userdir, 0o700)
```
Comment (mgedmin) [inline linkcheck/configuration/__init__.py]: This part is great!

File: linkcheck/decorators.py [hunk 2]
```diff
@@ -36,12 +36,16 @@ def h ():
...
 from __future__ import print_function
+
+import codecs
 import warnings
 import signal
 import os
 import sys
 import time
 
+from builtins import bytes
```
Comment (mgedmin) [inline linkcheck/decorators.py]: This part is a bit confusing.  (Doesn't Python 2 already have `bytes`?)

File: linkcheck/decorators.py [hunk 3]
```diff
@@ -125,9 +129,9 @@ def newfunc (*args, **kwargs):
         res = func(*args, **kwargs)
         duration = time.time() - t
         if duration > limit:
-            print(func.__name__, "took %0.2f seconds" % duration, file=log)
-            print(args, file=log)
-            print(kwargs, file=log)
+            log.write(bytes("%s took %0.2f seconds\n" %(func.__name__, duration), "utf8"))
+            log.write(bytes("%s\n" % str(args), "utf8"))
+            log.write(bytes("%s\n" % str(kwargs), "utf8"))
```
Comment (mgedmin) [inline linkcheck/decorators.py]: This part is very confusing?  What's going on here?

File: linkcheck/decorators.py [hunk 4]
```diff
@@ -36,12 +36,16 @@ def h ():
 
 """
 from __future__ import print_function
+
+import codecs
```
Comment (mgedmin) [inline linkcheck/decorators.py]: I don't see where this new import is getting used?

File: linkcheck/logger/blacklist.py [hunk 5]
```diff
@@ -90,7 +90,7 @@ def write_blacklist (self):
         """
         Write the blacklist.
         """
-        oldmask = os.umask(0077)
+        oldmask = os.umask(0o077)
```
Comment (mgedmin) [inline linkcheck/logger/blacklist.py]: This part is great!

File: tests/test_decorators.py [hunk 6]
```diff
@@ -42,4 +42,5 @@ def f ():
             time.sleep(1)
             return 42
         self.assertEqual(f(), 42)
-        self.assertTrue(log.getvalue())
+        log_value = log.getvalue()
+        self.assertEquals(log_value, b'f took 1.00 seconds\n()\n{}\n')
```
Comment (mgedmin) [inline tests/test_decorators.py]: assertEqual**s** emits a deprecation warning; please use `assertEqual`.

(And, again, I don't understand why this needs to be bytes?  Especially given that it's writing to a StringIO and not a BytesIO?)

Comment (cjmayo) [reply tests/test_decorators.py]: It was using:
`log = BytesIO()`
Changed to StringIO in python3_05c.

File: linkcheck/decorators.py (hunk 4 above)
Comment (cjmayo) [reply linkcheck/decorators.py]: Removed in python3_05c.

**Comment by cjmayo** [comment]
Split into #233 , #234 and #235.

**Pull request closed without merging**